src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,171 @@
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return searchByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode = searchByIndex(index);
+        T output = currentNode.value;
+        currentNode.value = value;
+        return output;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> output;
+        if (size == 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == 0) {
+            output = head;
+            if (head.next != null) {
+                head.next.prev = null;
+            }
+            head = head.next;
+            output.next = null;
+            size--;
+            return output.value;
+        }
+        if (size != 0 && index == size - 1) {
+            output = tail;
+            if (tail.prev != null) {
+                tail.prev.next = null;
+            }
+            tail = tail.prev;
+            output.prev = null;
+            size--;
+            return output.value;
+        }
+        Node<T> currentNode = searchByIndex(index);
+        output = currentNode;
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return output.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (currentNode.value == object
+                    || (currentNode.value != null && currentNode.value.equals(object))) {
+                remove(i);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private boolean isIndexOutOfArray(int index) {
+        return index >= size || index < 0;
+    }
+
+    public Node<T> searchByIndex(int index) {
+        if (isIndexOutOfArray(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> currentNode;
+        int startIndex;
+        if (index == 0) {
+            return head;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
+        if (index < size / 2) {
+            startIndex = 0;
+            currentNode = head;
+            while (startIndex < index) {
+                startIndex++;
+                currentNode = currentNode.next;
+            }
+            return currentNode;
+        }
+        startIndex = size - 1;
+        currentNode = tail;
+        while (startIndex > index) {
+            startIndex--;
+            currentNode = currentNode.prev;
+        }
+        return currentNode;
```

Comment:
let's implify that to checking whether after or before the middle and iterate using next or prev.

Reply:
done